tests/test_queue.py

```diff
@@ -1210,3 +1214,112 @@ async def test_workflow(var1: str) -> str:
         with SetWorkflowID(wfid2):
             handle2 = await queue.enqueue_async(test_workflow, "def")
     assert (await handle2.get_result()) == "def-c-p"
+
+    assert queue_entries_are_cleaned_up(dbos)
+
+
+def test_priority_queue(dbos: DBOS) -> None:
+    # Make sure that we can enqueue workflows with different priorities correctly
+    queue = Queue("test_queue_priority", 1)
+    child_queue = Queue("test_queue_child")
+
+    workflow_event = threading.Event()
+    wf_priority_list = []
+
+    @DBOS.workflow()
+    def child_workflow(p: int) -> int:
+        workflow_event.wait()
+        return p
+
+    @DBOS.workflow()
+    def test_workflow(priority: int) -> int:
+        wf_priority_list.append(priority)
+        # Make sure the priority is not propagated
+        assert assert_current_dbos_context().priority == None
+        child_handle = child_queue.enqueue(child_workflow, priority)
+        workflow_event.wait()
+        return child_handle.get_result() + priority
+
+    # Enqueue an invalid priority
+    with pytest.raises(Exception) as exc_info:
+        with SetEnqueueOptions(priority=-100):
+            queue.enqueue(test_workflow, -100)
+    assert "Invalid priority" in str(exc_info.value)
+
+    wf_handles = []
+    # First, enqueue a workflow without priority
+    handle = queue.enqueue(test_workflow, 0)
+    wf_handles.append(handle)
+
+    # Then, enqueue a workflow with priority 1 to 5
+    for i in range(1, 6):
+        with SetEnqueueOptions(priority=i):
+            handle = queue.enqueue(test_workflow, i)
+        wf_handles.append(handle)
```

Comment:
What happens if you enqueue all of these but the queue dequeues in the middle of the loop?

Reply:
It won't dequeue in the middle, because the workflow is blocked waiting for an event to be set. All workflows are in the queue before we call ` workflow_event.set()`.